src/main/java/online/bookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,73 @@
+package online.bookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import online.bookstore.dto.cart.AddCartItemDto;
+import online.bookstore.dto.cart.ShoppingCartDto;
+import online.bookstore.dto.cart.UpdateCartItemQuantityDto;
+import online.bookstore.model.User;
+import online.bookstore.service.ShoppingCartService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+@RestController
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @Operation(summary = "Add book to cart",
+            description = "Add book to cart and return cart info")
+    @PostMapping
+    @PreAuthorize("hasAnyRole('ROLE_USER')")
+    public ShoppingCartDto addBook(
+            @Valid @RequestBody AddCartItemDto addCartItemDto,
+            Authentication authentication
+    ) {
+        User user = (User) authentication.getPrincipal();
```

Comment:
You can create separate method to avoid duplication of code to get `Long userId `